core/src/impl/Kokkos_Core.cpp

```diff
@@ -166,9 +205,7 @@ int get_ctest_gpu(const char* local_rank_str) {
   return std::stoi(id.c_str());
 }
 
-namespace {
-
-#if defined(KOKKOS_ENABLE_CUDA) || defined(KOKKOS_ENABLE_HIP)
+// extern function to extract gpu # from args
```

Comment:
I don't understand the function is defined here so it's not extern.
```suggestion
// function to extract gpu # from args
```